test/tstScopedProfileRegion.cpp

```diff
@@ -0,0 +1,94 @@
+/****************************************************************************
+ * Copyright (c) 2017-2021 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX_DetailsKokkosExtScopedProfileRegion.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+#include <stack>
+#include <string>
+#include <type_traits>
+
+#if (KOKKOS_VERSION >= 30200) // callback registration from within the program
+                              // was added in Kokkkos v3.2
+
+namespace
+{
+std::stack<std::string> arborx_test_region_stack;
+
+// NOTE: cannot use lambdas because they can only be converted to function
+// pointers if they don't capture anything
+void arborx_test_push_region(char const *label)
+{
+  BOOST_TEST_MESSAGE(std::string("push ") + label);
+  arborx_test_region_stack.push(label);
+}
+
+void arborx_test_pop_region()
+{
+  auto const &label = arborx_test_region_stack.top();
+  BOOST_TEST_MESSAGE(std::string("pop ") + label);
+  arborx_test_region_stack.pop();
+}
+
+} // namespace
+
+BOOST_AUTO_TEST_SUITE(KokkosToolsAnnotations)
+
+BOOST_AUTO_TEST_CASE(scoped_profile_region)
+{
+  Kokkos::Tools::Experimental::set_push_region_callback(
+      arborx_test_push_region);
+  Kokkos::Tools::Experimental::set_pop_region_callback(arborx_test_pop_region);
+
+  BOOST_TEST(arborx_test_region_stack.empty());
+
+  KokkosExt::ScopedProfileRegion("bug"); // unnamed guard!
```

Comment:
The comment confuses me. First, does "unnamed" mean that it's not prefixed "ArborX::"? Second, what is the reason behind doing it this way?

Reply:
Not declaring a variable, the region is popped at the end of the statement.

Reply:
OK, so unnamed in this context means "not a variable"? Because it already caused confusion, I ask to explicitly state that.

Reply:
I added a comment.  Let me know if that is still not clear.

Just illustrating how not to use the scope guard.